onnxruntime/core/session/inference_session.cc

```diff
@@ -492,12 +492,25 @@ common::Status InferenceSession::RegisterExecutionProvider(const std::shared_ptr
   if (provider_type == onnxruntime::kDmlExecutionProvider) {
     // DML's memory is not byte addressable and hence mem pattern doesn't work.
     if (session_options_.enable_mem_pattern) {
-      LOGS(*session_logger_, WARNING)
-          << "Having memory pattern enabled is not supported while using the DML Execution Provider. "
-          << "So disabling it for this session since it uses the DML Execution Provider.";
       session_options_.enable_mem_pattern = false;
     }
 
+    // Default this option to true when the DML EP is registered.
+    // This should be removed if QDQ is supported for DML through QDQSelectorActionTransformer and the DML EP does not 
+    // rely on the constant folding pass for DequantizeLinear.
+    optional<std::string> disable_quant_qdq = session_options_.config_options.GetConfigEntry(kOrtSessionOptionsDisableQuantQDQ);
+
+    if (disable_quant_qdq == std::nullopt) {
+      auto st = session_options_.config_options.AddConfigEntry(kOrtSessionOptionsDisableQuantQDQ, "1");
```

Comment:
We don't usually modify session options on behalf of users as it's not what the user created the session with. If this is absolutely required, we should log a message that says so. Logging will also help with debugging scenarios where this might turn out to be non-optimal.

Reply:
The "enable_mem_pattern" option has the same issue.

Reply:
Default value is 0: https://github.com/microsoft/onnxruntime/blob/eb8f6c7c52fa7ffd3e9b6865a7646f8f243a124b/include/onnxruntime/core/session/onnxruntime_session_options_config_keys.h#L44-L48